airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -173,6 +176,26 @@ private WebBackendConnectionRead buildWebBackendConnectionRead(final ConnectionR
       webBackendConnectionRead.setLatestSyncJobStatus(job.getStatus());
     });
 
+    SchemaChange schemaChange = SchemaChange.NO_CHANGE;
+
+    if (connectionRead.getSourceId() != null && currentSourceCatalogId.isPresent()) {
+      final Optional<ActorCatalogFetchEvent> mostRecentFetchEvent =
+          configRepository.getMostRecentActorCatalogFetchEventForSource(connectionRead.getSourceId());
+
+      if (mostRecentFetchEvent.isPresent()) {
+        final ActorCatalog currentCatalog = configRepository.getActorCatalogById(currentSourceCatalogId.get());
```

Comment:
I think this fetch is not necessary because we just use the catalog Id from the currentSourceCatalogId and this is a foreign key so there is no risk that it doesn't exist.

Reply:
👍 made that change